src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -123,7 +138,41 @@ RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::RadialBasisFctMapping(
     setInputRequirement(Mapping::MeshRequirement::VERTEX);
     setOutputRequirement(Mapping::MeshRequirement::VERTEX);
   }
+
+  PRECICE_CHECK(polynomial == Polynomial::ON || polynomial == Polynomial::SEPARATE, "RBF mappings without a polynomial ( polynomial = \"off\") are only implemented using PETSc. Please install preCICE with PETSc or remove the use-qr-decomposition=\"true\" tag in your configuration file.");
   setDeadAxis(xDead, yDead, zDead);
+
+  // Count number of dead dimensions
+  int deadDimensions = 0;
+  for (int d = 0; d < dimensions; d++) {
+    if (_deadAxis[d])
+      deadDimensions += 1;
+  }
+  polyparams    = (_polynomial == Polynomial::ON) ? 1 + dimensions - deadDimensions : 0;
+  sepPolyparams = (_polynomial == Polynomial::SEPARATE) ? 1 + dimensions - deadDimensions : 0;
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::setDeadAxis(bool xDead, bool yDead, bool zDead)
+{
+  // TODO: Remove duplication here with PETSc based implementation
```

Comment:
This could be done by a common abstract basis class. Or do you have another solution in mind?
These "TODOs" in the code get always easily lost. Better to open an issue (here "good first issue") and remove the TODO here.